addons/sale_management/models/sale_order.py

```diff
@@ -20,6 +20,13 @@ class SaleOrder(models.Model):
         copy=True, readonly=True,
         states={'draft': [('readonly', False)], 'sent': [('readonly', False)]})
 
+    @api.constrains('company_id', 'sale_order_option_ids')
```

Comment:
Same here
`sale_order_option_ids` --> `sale_order_option_ids.product_id`

Reply:
Constrains decorator doesn't support dotted names, so it should stay like this ;).
https://www.odoo.com/documentation/12.0/reference/orm.html#odoo.api.constrains

Reply:
I was not aware of this.
Thanks